src/modules/SystemSettings.rb

```diff
@@ -15,210 +15,304 @@ module Yast
+      # (which is imported by the Bootloader (imported by the SystemSettings
+      # module)) has a Read() function call in its constructor.
+      Yast.import "Bootloader"
 
-        @elevator = io_scheduler
-      else
-        Builtins.y2error(
-          "unknown IO scheduler '%1', use: %2",
-          io_scheduler,
-          GetPossibleElevatorValues()
-        )
-      end
+      # Read bootloader settings in normal mode
+      Bootloader.Read if Mode.normal
 
-      nil
+      # Set IO scheduler
+      SetIOScheduler(current_elevator)
+      Builtins.y2milestone("Global IO scheduler: %1", GetIOScheduler())
     end
 
-    def GetSysRqKeysEnabled
-      @ENABLE_SYSRQ != nil && @ENABLE_SYSRQ != "0"
+    # Read SysRq keys configuration updating the module's value
+    #
+    # @see Read
+    def read_sysrq
+      if kernel_sysrq != sysctl_sysrq
+        log.warn(
+          format("SysRq mismatch: sysconfig value: '%s', current: '%s'", sysctl_sysrq, kernel_sysrq)
+        )
+      end
+
+      # display the current value if it not configured
+      self.enable_sysrq = sysctl_sysrq || kernel_sysrq
     end
 
-    def SetSysRqKeysEnabled(enable_sysrq)
-      if enable_sysrq == nil
-        Builtins.y2warning("enable_sysrq should be 'true' or 'false'")
+    # Write SysRq keys settings
+    #
+    # @see Write
+    def write_sysrq
+      if enable_sysrq.nil? || enable_sysrq !~ /\A[0-9]\z/
+        log.warn("Not writing invalid ENABLE_SYSRQ value: #{enable_sysrq}")
         return
       end
 
-      enable_sysrq_string = enable_sysrq ? "1" : "0"
-
-      @modified = true if @ENABLE_SYSRQ != enable_sysrq_string
-
-      @ENABLE_SYSRQ = enable_sysrq_string
-      Builtins.y2milestone("SysRq was set to %1", @ENABLE_SYSRQ)
-
-      nil
+      log.info("Saving ENABLE_SYSRQ: #{enable_sysrq}")
+      SCR.Write(path(".etc.sysctl_conf.\"kernel.sysrq\""), enable_sysrq)
+      SCR.Write(path(".etc.sysctl_conf"), nil)
     end
 
-    publish :function => :GetPossibleElevatorValues, :type => "list <string> ()"
-    publish :function => :Modified, :type => "boolean ()"
-    publish :function => :Read, :type => "boolean ()"
-    publish :function => :Activate, :type => "boolean ()"
-    publish :function => :Write, :type => "boolean ()"
-    publish :function => :GetIOScheduler, :type => "string ()"
-    publish :function => :SetIOScheduler, :type => "void (string)"
-    publish :function => :GetSysRqKeysEnabled, :type => "boolean ()"
-    publish :function => :SetSysRqKeysEnabled, :type => "void (boolean)"
+    # Write IO Scheduler settings
+    #
+    # This method only has effect during normal mode. During installation,
+    # bootloader configuration is written at the end of the first stage.
+    #
+    # @see Bootloader#Write
+    # @see Write
+    def write_scheduler
+      return true unless Mode.normal
+      Yast.import "Bootloader"
+      Bootloader.Write
+      true
+    end
```

Comment:
in general whole this class looks to me like it deserve two objects: Scheduler and Sysrq one, with its read/write/activating/etc.